### PR TITLE
Add getState to route resolve to allow the use of theme loading

### PR DESCRIPTION
**Pull request by orionstein**
This lets us use getState in route resolves to load and unload our themes. This needs to happen in the resolve because we want to load these before the route changes and we get new components

### CODE REVIEW

File: src/transition.js [hunk 1]
```diff
@@ -83,7 +85,17 @@ export default async (options = {}) => {
...
-    const response = Promise.resolve(resolver({ route, params, location }))
+    const response = Promise.resolve(
+      resolver({
+        route,
+        params,
+        location,
+        // This has to be done because we get getState on initial match
+        // And store on subsequent client transitions
+        // At some point we could switch it to use store fully
+        getState: getState || (store || { getState: () => ({}) }).getState,
```
Comment (lfender6445) [inline src/transition.js]: `(store || { getState: () => ({}) })` is this for tests?

Comment (orionstein) [reply src/transition.js]: This is for adding support for getState to our resolve functions

Comment (orionstein) [reply src/transition.js]: Transition, match, and initialize all are initialized slightly differently

Comment (barelyfitz) [inline src/transition.js]: This line is a little hard to parse, maybe make it more readable?
```
getState: getState || get(store, 'getState') || () => ({})
```
